nucypher/blockchain/eth/clients.py

```diff
@@ -229,27 +243,35 @@ def sync(self,
         now = maya.now()
         start_time = now
 
+
         def check_for_timeout(t):
             last_update = maya.now()
             duration = (last_update - start_time).seconds
             if duration > t:
                 raise self.SyncTimeout
 
-        # Check for ethereum peers
-        self.log.info(f"Waiting for Ethereum peers ({len(self.peers)} known)")
-        while not self.peers:
-            time.sleep(0)
-            check_for_timeout(t=60)
-
-        # Wait for sync start
-        self.log.info(f"Waiting for {self.chain_name.capitalize()} chain synchronization to begin")
-        while not self.syncing:
-            time.sleep(0)
-            check_for_timeout(t=120)
-
-        while self.syncing:
-            self.log.info(f"Syncing {self.syncing['currentBlock']}/{self.syncing['highestBlock']}")
-            time.sleep(5)
+        while not self._has_latest_block():
+            # Check for ethereum peers
+            self.log.info(f"Waiting for Ethereum peers ({len(self.peers)} known)")
+            while not self.peers:
+                time.sleep(0)
+                check_for_timeout(t=self.PEERING_TIMEOUT)
```

Comment:
@vepkenez did you intend for this timeout to go from a 60s timeout previously to a 5s timeout now (self.PEERING_TIMEOUT)?

Reply:
I don't think this timeout stuff is working at all right now.  Still working on that.